### ISSUE TITLE
Cannot Open Server-side Code Since 1.4.2

### ISSUE THREAD

**Issue by isc-jsmith** [intersystems-community/vscode-objectscript#879]
Issue Type: <b>Bug</b>

Since installing 1.4.2 when trying to open a serverside copy of the class using the ObjectScript Explorer, I consistently get a messages informing me that, "The editior could not be opened because the file was not found". Copying the file to my local workspace allows me to open it though.

Extension version: 1.4.2
VS Code version: Code 1.64.2 (f80445acd5a3dadef24aa209168452a3d97cc326, 2022-02-09T22:00:58.347Z)
OS version: Darwin x64 20.6.0
Restricted Mode: No

**Comment by isc-bsaviano** [comment]
@isc-jsmith What are the steps I can do to reproduce this? I'm on mac as well and the ObjectScript Explorer using 1.4.2 is working fine for me.

**Comment by isc-jsmith** [comment]
I just opened an existing workspace and tried open a file using the ObjectScript explorer. I can browse the tree of classes without issue. This seems to happen on both IRIS 2021.1 and IRIS 2020.1. What is interesting is that it doesn't even seem to be hitting Apache to call the Atelier API.

**Comment by isc-bsaviano** [comment]
Do you see any error messages anywhere? Anything in the ObjectScript Output channel?

**Comment by isc-jsmith** [comment]
Nothing in the ObjectScript Output channel. Tried with JSON output enabled as well. I do the get this in Log (Window) output:

`[2022-02-25 07:18:34.711] [renderer4] [error] Error: File not found
    at S.handleSetInputError (vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:2351:107078)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async S.setInput (vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:2351:106753)
    at async i.doSetInput (vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:2351:26831)
    at async i.doOpenEditor (vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:2351:25204)
    at async i.openEditor (vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:2351:24830)
    at async vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:2551:16600
    at async vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:2351:47636`

**Comment by isc-bsaviano** [comment]
I don't that log info is related to our extension. Do you have the `objectscript.serverSideEditing` setting enabled?

**Comment by isc-jsmith** [comment]
I do have that setting enabled. Interesting that message above seems to occur immediately after trying to open a server side file. I also get a message saying, for example:

> Unable to open 'CachedObject.cls': File not found.

Where "CachedObject.cls" is the class name without the package(s).

**Comment by isc-bsaviano** [comment]
With that setting on I can reproduce your issue.

**Comment by isc-jsmith** [comment]
Oh. This seems to occur if I have "Server Side Editing" on. I can open the files read only with it disabled.

Here are the relevant sections from the Workspace's settings.json:
```
    "objectscript.autoPreviewXML": true,
    "objectscript.export": {

        "folder": "src",
        "addCategory": true,
        "map": {},
        "atelier": true,
        "generated": false,
        "filter": "",
        "category": "*",
        "noStorage": false,
        "dontExportIfNoChanges": false,
        "maxConcurrentConnections": 0
    },
    "objectscript.multilineMethodArgs": true,
    "objectscript.studioActionDebugOutput": true,
    "objectscript.compileFlags": "cuk /compileembedded=1",
    "objectscript.serverSourceControl.disableOtherActionTriggers": true,
    "objectscript.serverSideEditing": true
```

**Comment by isc-bsaviano** [comment]
Thanks for that. Do you remember the last version this was working with?

**Comment by isc-jsmith** [comment]
It seemed to work fine with v1.4.1
________________________________

**Comment by isc-bsaviano** [comment]
I'm surprised because none of this code has changed recently. It appears that the URI's being generated when `serverSideEditing` is `true` are junk so I would expect this problem to have manifested by now.

**Comment by isc-jsmith** [comment]
I've gone back to 1.4.1 and it seems to be working okay.

**Comment by a-boertien** [comment]
I'm seeing an error that may be related
I have a multi folder workspace, with 2 folders for server side ObjectScript files & web files

It is no longer connecting to the web files folder in 1.4.2

On the web folder in explorer, I get a warning icon and "Unable to resolve workspace folder" tooltip

The Log (Window) output shows:

> [2022-02-24 17:42:27.016] [renderer1] [warning] Ignoring the error while validating workspace folder isfs://local:dev/path/to/web?csp - Unable to resolve nonexistent file 'isfs://local:dev/path/to/web?csp'

I've also switched back to 1.4.1, which is working

**Comment by isc-bsaviano** [comment]
@gjsjohnmurray Is there anything I'm missing here? I don't see what changed between  1.4.1 and 1.4.2 that could be causing this.

**Comment by gjsjohnmurray** [comment]
A bisect with the 1.4.2 betas led to this having broken in beta.12 (PR #857 from me). I am investigating.

**Comment by gjsjohnmurray** [comment]
@isc-jsmith @a-boertien please try the dev vsix from the zip at https://github.com/intersystems-community/vscode-objectscript/suites/5444562033/artifacts/172865984

**Comment by a-boertien** [comment]
@gjsjohnmurray I tried and it is working (for the error I was seeing)

**Comment by gjsjohnmurray** [comment]
@a-boertien  thanks for confirming. PR has merged and a new 1.4.3 beta will be available shortly at https://github.com/intersystems-community/vscode-objectscript/releases

We plan to publish 1.4.3 early next week.

**Comment by isc-jsmith** [comment]
Fixed the issue I was seeing as well.
